### PR TITLE
Add a .unitypackage for MRTK's TestUtilities

### PR DESCRIPTION
**Pull request by wiwei**
The TestUtility refactor that @CDiaz-MS did a while back added some cool NuGet package for the MRTK test infrastructure. This has been used internally by a few teams to build out their test infra, and up until now we haven't had a .unitypackage equivalent.

This PR publishes the core set of test infrastructure that the MRTK itself uses to write integration tests, things like:

1. Show hand there.
2. Pinch
3. Move hand over there.
4. Release

This lets you write automated tests that interact with things in your scene, so that instead of having to do manual test passes, you can have automated (i.e. CI-time) coverage of core/common things. The tests that we've written have helped us avoid regressions in components and to prove functionality, and it's like that consumers will also be able to benefit from this.

Note that there is an initial investment to writing any set of tests, but the payoff is usually worth it (it has been worth it for us)

### CODE REVIEW

File: Documentation/ReleaseNotes.md [hunk 1]
```diff
@@ -7,6 +7,30 @@
...
+yield return leftHand.Move(new Vector3(0.2f, 0.2f, 0));
+yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+```
+
+For instructions on how to write a test using these TestUtilities, see this section on
+[writing tests](Contributing/UnitTests.html#writing-tests)
+
+For examples of existing tests that use this infrastructure, see MRTK's [PlayModeTests]
+(https://github.com/microsoft/MixedRealityToolkit-Unity/tree/mrtk_development/Assets/MRTK/Tests/PlayModeTests)
+
```
Comment (CDiaz-MS) [inline Documentation/ReleaseNotes.md:33]: There is a space which displays the link incorrectly 

```suggestion
For examples of existing tests that use this infrastructure, see MRTK's [PlayModeTests](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/mrtk_development/Assets/MRTK/Tests/PlayModeTests)

```

**Comment by davidkline-ms** [comment]
I'd recommend moving the test utilities into the tools folder / package rather than creating a new package. this is where the upm package proposal is considering placing it.

**Comment by wiwei** [comment]
@davidkline-ms this is consistent with the current set of nuget packages, so I'd rather keep this as-is.

Also, I don't believe that this is a "Tool" (i.e. it's not a UI/Editor tool). We should separately figure out what the long term packages are and then do a one time pass to bring things along.

**Comment by wiwei** [comment]
More generally, to reduce complexity/overhead we should have the same set of packages regardless of distribution method (i.e. what we group up is the same, HOW we package them can be different i.e. UPM/.unitypackage, etc)

File: Documentation/ReleaseNotes.md [hunk 2]
```diff
@@ -7,6 +7,29 @@
...
+```csharp
+TestHand leftHand = new TestHand(Handedness.Left);
+yield return leftHand.Show(new Vector3(-0.1f, -0.1f, 0.5f));
+yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+yield return leftHand.Move(new Vector3(0.2f, 0.2f, 0));
+yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+```
+
+For instructions on how to write a test using these TestUtilities, see this section on
+[writing tests](Contributing/UnitTests.html#writing-tests)
```
Comment (wiwei) [inline Documentation/ReleaseNotes.md:29]:
```suggestion
[writing tests](Contributing/UnitTests.md#writing-tests)
```

**Comment by davidkline-ms** [comment]
> More generally, to reduce complexity/overhead we should have the same set of packages regardless of distribution method (i.e. what we group up is the same, HOW we package them can be different i.e. UPM/.unitypackage, etc)

Agreed.

**Pull request merged**